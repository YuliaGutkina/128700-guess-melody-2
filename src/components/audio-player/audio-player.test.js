import React from "react";
import renderer from "react-test-renderer";
import {AudioPlayer} from "./audio-player";

function createNodeMock(element) {
  if (element.type === `audio`) {
    return {
      focus() {},
    };
  }
  return null;
}

it(`Genre Screen correctly renders after relaunch`, () => {
  const options = {createNodeMock};
  const tree = renderer
    .create(
        <AudioPlayer
          isPlaying={false}
          onPlayButtonClick={jest.fn()}
          src={`https://upload.wikimedia.org/wikipedia/commons/6/64/Ugandan_national_anthem%2C_performed_by_the_U.S._Navy_Band.ogg`}
        />, options)
    .toJSON();

  expect(tree).toMatchSnapshot();
});
