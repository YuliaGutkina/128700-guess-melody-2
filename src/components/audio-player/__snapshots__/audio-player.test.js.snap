// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Genre Screen correctly renders after relaunch 1`] = `
Array [
  <button
    className="track__button track__button--play"
    disabled={true}
    onClick={[Function]}
    type="button"
  />,
  <div
    className="track__status"
  >
    <audio
      src="https://upload.wikimedia.org/wikipedia/commons/6/64/Ugandan_national_anthem%2C_performed_by_the_U.S._Navy_Band.ogg"
    />
  </div>,
]
`;
